java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCSToJDBC.java

```diff
@@ -47,13 +54,39 @@ public static GCSToJDBC of(String... args) {
   public void runTemplate() {
     validateInput();
     try (SparkSession spark = SparkSession.builder().appName("GCS to JDBC").getOrCreate()) {
-      Dataset<Row> dataset =
-          spark.read().format(config.getInputFormat()).load(config.getInputLocation());
-
+      Dataset<Row> dataset = load(spark);
       write(dataset);
     }
   }
 
+  public Dataset<Row> load(SparkSession spark) {
+    Dataset<Row> inputData = null;
+
+    switch (config.getInputFormat()) {
+      case GCS_JDBC_CSV_FORMAT:
+        inputData =
+            spark
+                .read()
+                .format(config.getInputFormat())
+                .option(GCS_JDBC_CSV_HEADER, true)
+                .option(GCS_JDBC_CSV_INFER_SCHEMA, true)
```

Comment:
If we are now entertaining CSV as well, then we should also add "delimiter" as input too.
In addition there can be many more other configurable options like (dateFormat, nullValue, escape etc).
This is open for discussion.

Taking clues from respective Dataflow template.
https://cloud.google.com/dataflow/docs/guides/templates/provided-batch#gcs_text_to_cloud_spanner

Reply:
In keeping with trying to be consistent with other templates I only added the same two hardcoded options. I agree these these should be configurable and we should look to add more control. But I think that is something to be implemented across all CSV related templates and not just this one.
I believe it should be an issue in its own right.

Reply:
Agreed verbally that there should be another issue raised for full CSV option control across all relevant templates.

Reply:
+1